analyzers/tests/SonarAnalyzer.UnitTest/TestCases/NonStandardCryptographicAlgorithmsShouldNotBeUsed.vb

```diff
@@ -0,0 +1,350 @@
+﻿Imports System
+Imports System.Collections.Generic
+Imports System.Linq
+Imports System.Text
+Imports System.Security.Cryptography
+
+Public Class CustomHash  ' Noncompliant {{Make sure using a non-standard cryptographic algorithm is safe here.}}
+'            ^^^^^^^^^^
+    Inherits HashAlgorithm
+
+    Public Overrides Sub Initialize()
+        Throw New NotImplementedException()
+    End Sub
+
+    Protected Overrides Sub HashCore(array() As Byte, ibStart As Integer, cbSize As Integer)
+        Throw New NotImplementedException()
+    End Sub
+
+    Protected Overrides Function HashFinal() As Byte()
+        Throw New NotImplementedException()
+    End Function
+
+End Class
+
+Public class CustomCryptoTransform  ' Noncompliant {{Make sure using a non-standard cryptographic algorithm is safe here.}}
+'            ^^^^^^^^^^^^^^^^^^^^^
+    Implements ICryptoTransform
+
+    Private disposedValue As Boolean
+
+    Public ReadOnly Property InputBlockSize As Integer Implements ICryptoTransform.InputBlockSize
+        Get
+            Throw New NotImplementedException()
+        End Get
+    End Property
+
+    Public ReadOnly Property OutputBlockSize As Integer Implements ICryptoTransform.OutputBlockSize
+        Get
+            Throw New NotImplementedException()
+        End Get
+    End Property
+
+    Public ReadOnly Property CanTransformMultipleBlocks As Boolean Implements ICryptoTransform.CanTransformMultipleBlocks
+        Get
+            Throw New NotImplementedException()
+        End Get
+    End Property
+
+    Public ReadOnly Property CanReuseTransform As Boolean Implements ICryptoTransform.CanReuseTransform
+        Get
+            Throw New NotImplementedException()
+        End Get
+    End Property
+
+    Public Function TransformBlock(inputBuffer() As Byte, inputOffset As Integer, inputCount As Integer, outputBuffer() As Byte, outputOffset As Integer) As Integer Implements ICryptoTransform.TransformBlock
+        Throw New NotImplementedException()
+    End Function
+
+    Public Function TransformFinalBlock(inputBuffer() As Byte, inputOffset As Integer, inputCount As Integer) As Byte() Implements ICryptoTransform.TransformFinalBlock
+        Throw New NotImplementedException()
+    End Function
+
+    Protected Overridable Sub Dispose(disposing As Boolean)
+        If Not disposedValue Then
+            If disposing Then
+                ' TODO: dispose managed state (managed objects)
+            End If
+
+            ' TODO: free unmanaged resources (unmanaged objects) and override finalizer
```

Comment:
we should remove the generated code for dispose and add throw instead

Edit: check also the others